Refresh.GameServer/Types/Assets/GameAsset.cs

```diff
@@ -23,4 +32,341 @@ public partial class GameAsset : IRealmObject
+        }
+        
+        Rectangle cropRectangle;
+        
+        //If the image is wider than it is tall
+        cropRectangle = image.Width > image.Height ? new Rectangle(image.Width / 2 - image.Height / 2, 0, image.Height, image.Height) :
+            //If the image is taller than it is wide
+            new Rectangle(0, image.Height / 2 - image.Width / 2, image.Width, image.Width);
+
+        int targetWidth = Math.Clamp(cropRectangle.Width, 16, maxWidth);
+        
+        //Round to the nearest multiple of 16, this is to make PSP happy
+        targetWidth = 
+            (int)Math.Round(
+                targetWidth / (double)16,
+                MidpointRounding.AwayFromZero
+            ) * 16;
+        
+        copy.Mutate(ctx => ctx.Crop(cropRectangle).Resize(targetWidth, targetWidth));
+        
+        return copy;
+    }
+
+    /// <summary>
+    /// Transforms an image to be consumed by a particular game
+    /// </summary>
+    /// <param name="game">The game which will consume the resulting asset</param>
+    /// <param name="dataStore">The data store</param>
+    /// <param name="decodeImage">The function used to decode an image from the data store</param>
+    /// <param name="transformImage">The transformation function</param>
+    /// <returns>The hash of the transformed asset</returns>
+    /// <exception cref="NotImplementedException">That conversion step is unimplemented at the moment</exception>
+    /// <exception cref="ArgumentOutOfRangeException">Invalid TokenGame</exception>
+    private string TransformImage(TokenGame game, IDataStore dataStore, Func<string, Image<Rgba32>> decodeImage, Func<Image, Image<Rgba32>?> transformImage)
+    {
+        string dataStorePath = this.IsPSP ? $"psp/{this.AssetHash}" : this.AssetHash;
+
+        bool mainlineDoesntNeedConversion = this.AssetType is GameAssetType.Png or GameAssetType.Texture or GameAssetType.GameDataTexture;
+        
+        switch (game)
+        {
+            case TokenGame.Website:
+            case TokenGame.LittleBigPlanet1: 
+            case TokenGame.LittleBigPlanet2:
+            case TokenGame.LittleBigPlanet3:
+            case TokenGame.LittleBigPlanetVita: {
+                Image sourceImage = decodeImage(dataStorePath);
+                
+                //Load the image from the data store and transform it
+                Image? image = transformImage(sourceImage);
+                //If its null, then no transformation was needed
+                if (image == null)
+                {
+                    if (mainlineDoesntNeedConversion)
+                    {
+                        //Return the existing asset hash
+                        return this.AssetHash;
+                    }
+
+                    //Set the image to use to the source image
+                    image = sourceImage;
+                }
+
+                //Save the image as a PNG file in a byte array in memory
+                MemoryStream memory = new();
```

Comment:
That variable name makes me think `Memory<T>`, not `MemoryStream`. In refresh we usually use `ms` or just `stream`.